src/keystore.rs

```diff
@@ -11,8 +11,11 @@ pub trait Keystore {
     type KeyId;
 
     /// Public key for a given id
-    fn public(&self, id: Self::KeyId) -> Result<Borrown<'_, PublicKey>, Self::Error>;
+    fn public(&self, id: &Self::KeyId) -> Result<Borrown<'_, PublicKey>, Self::Error>;
+
+    /// Return `()` if the provided key exists in the storage
+    fn public_identity_exists(&self, public: &PublicKey) -> Result<(), Self::Error>;
```

Comment:
I wouldn't consider it an error if it doesn't exist. Seems like this should be `Result<bool, Self::Error>`, since Error should be reserved for a system issue or unexpected problem, whereas it might be expected for a public key to not exist in some cases from an API consumer point of view.

Reply:
yes, agree. TBH I'm not sure if this function should exist in the trait. I'm making some tests; will set this PR to draft